src/main/kotlin/org/rust/ide/presentation/TypeRendering.kt

```diff
@@ -266,15 +257,30 @@ private data class TypeRenderer(
     private fun formatBoundElementGenerics(
         boundElement: BoundElement<RsGenericDeclaration>,
         render: (Ty) -> String
+    ): List<String> = formatGenerics(boundElement.element, boundElement.subst, render)
+
+    private fun formatGenerics(
+        declaration: RsGenericDeclaration,
+        subst: Substitution,
+        render: (Ty) -> String
     ): List<String> {
-        val tySubst = boundElement.element.typeParameters.map { render(boundElement.subst[it] ?: TyUnknown) }
-        val regionSubst = if (includeLifetimeArguments) {
-            boundElement.element.lifetimeParameters.map { render(boundElement.subst[it] ?: ReUnknown) }
-        } else {
-            emptyList()
-        }
-        val constSubst = boundElement.element.constParameters.map { render(boundElement.subst[it] ?: CtUnknown) }
-        return regionSubst + tySubst + constSubst
+        var nonDefaultTyFound = false
+        return declaration.genericParameters
```

Comment:
nit: I personally prefer not to use `filter`, `map`, etc. methods when they have some state (`nonDefaultTyFound` here) and use simple `for` loop instead. Also, it allows not to create a separate list object by `map`. But it's up to you